internal/commands/result_test.go

```diff
@@ -330,5 +346,45 @@ func TestSBOMReportXMLWithProxy(t *testing.T) {
 func TestRunGetResultsByScanIdGLFormat(t *testing.T) {
 	execCmdNilAssertion(t, "results", "show", "--scan-id", "MOCK", "--report-format", "gl-sast")
 	// Run test for gl-sast report type
-	os.Remove(fmt.Sprintf("%s.%s", fileName, printer.FormatGL))
+	removeFile(t, "cx_result.gl-sast-report", "json")
+}
+func TestRunResultsShow_ContainersFFIsOn_includeContainersResult(t *testing.T) {
+	mock.Flags = wrappers.FeatureFlagsResponseModel{{Name: wrappers.ContainerEngineCLIEnabled, Status: true}}
+	execCmdNilAssertion(t, "results", "show", "--scan-id", "MOCK", "--report-format", "json")
+	assertContainersPresent(t, true)
+	// Remove generated json file
+	removeFileBySuffix(t, printer.FormatJSON)
+}
+func TestRunResultsShow_ContainersFFIsOff_excludeContainersResult(t *testing.T) {
+	mock.Flags = wrappers.FeatureFlagsResponseModel{{Name: wrappers.ContainerEngineCLIEnabled, Status: false}}
+	execCmdNilAssertion(t, "results", "show", "--scan-id", "MOCK", "--report-format", "json")
+	assertContainersPresent(t, false)
+	// Remove generated json file
+	removeFileBySuffix(t, printer.FormatJSON)
+}
+func TestRunResultsShow_AgentIsNotSupported_excludeContainersResult(t *testing.T) {
+	mock.Flags = wrappers.FeatureFlagsResponseModel{{Name: wrappers.ContainerEngineCLIEnabled, Status: true}}
+	execCmdNilAssertion(t, "results", "show", "--scan-id", "MOCK", "--report-format", "json", "--agent", "jet-brains")
+	assertContainersPresent(t, false)
+	// Remove generated json file
+	removeFileBySuffix(t, printer.FormatJSON)
+}
+
+func assertContainersPresent(t *testing.T, hasContainersPresent bool) {
+	bytes, err := os.ReadFile(fileName + "." + printer.FormatJSON)
+	assert.NilError(t, err, "Error reading file")
+	// Unmarshal the JSON data into the ScanResultsCollection struct
+	var scanResultsCollection *wrappers.ScanResultsCollection
+	err = json.Unmarshal(bytes, &scanResultsCollection)
+	assert.NilError(t, err, "Error unmarshalling JSON data")
+	for _, scanResult := range scanResultsCollection.Results {
+		if !hasContainersPresent && scanResult.Type == params.ContainersType {
```

Comment:
hasContainersPresent == isContainerEnabled? if so change the name, it's not clear..